compiler/hash-source/src/constant.rs

```diff
@@ -20,12 +20,22 @@ use crate::identifier::{Identifier, CORE_IDENTIFIERS};
 #[derive(Debug, Clone, Copy)]
 pub struct FloatConstant {
     /// Raw value of the float
-    pub value: f64,
+    pub value: [u8; 8],
```

Comment:
Why not store as f64?

Reply:
When the value is `f32`, it would inappropriate to store it as an `f64`. Both variants should be stored as a byte scalar, which is then appropriately used as required. This is particularly important for operations that occur on the constants, later on in the compilation stage.

Reply:
Why not `enum FloatConstant { F32(f32), F64(f64) }` ?

Reply:
I think this is a better approach for now, however it would be nice to later unify all these stores to just use a scalar. Especially in the future when `constructors` are also stored as constants.

Reply:
How would constructors be stored as constants? I agree about the unification but not needed yet.

Reply:
Constructors should be stored as a byte scalar which has an attached layout of how it is represented.

Reply:
But that is only relevant at IR? This is parsing related no?